### PR TITLE
Burn the Old World Down to Make Way For the New: The Economy

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Our economy is broken, to say the least. It's utterly and completely pointless and serves no purpose, what so ever at all.

This completely and UTTERLY guts it from the game.

Now hold up, before you get too excited, please read all of this before commenting or reacting.

There was an old coder known as Cael Aislinn...his code was *legendarily* bad in Maintainers eyes and other coders area also aware of his status (@Aurorablade can definitely attest). Needless to say "Cael Code" was a well known insult and complaint, back in the day. Most of his code has been purged---just about all of it, in fact----but one thing remains: Economy code.

The way this is designed is horrendous, to say the least---it has a lot of complicated, poorly coded moving parts, and interacting with the system is inconvenient, at best---worse yet, the only thing it's reallly used for it metagame the heck out of changelings and vaguely act as a limiter for the "merch" items near cargo.

Worry not though, as the title implies, a new world *can* be built. I'll be outlining, in a document, TG's economy overhaul, what it means, why it's better, and make a post about it, which I will link here. Needless to say, it ties the station together in a wonderful way, essentially gives everyone objectives to do, forces scarcity of items, thus creating interesting and dynamic choices, and fundamentally alters how cargo interacts with the rest of the station---it not only has massive roleplay potential, but it's mechanically well designed, too---it's far more convenient for players to use while still functioning as an *actual* economy.

Why not implement that here? It's a complicated system, and ripping this out while simultaneously putting that in would be extremely difficult and hard to analyze, especially since a lot of variables have *similar* names and behaviors. Let's be frank, too, nothing is really being lost here---money is almost *never* utilized on a serious basis, and no one really cares about it; there's vague inclinations of having lots of it, to stroke your own ego, but it doesn't really accomplish anything---nor is it even utilized in RP.

So, burn it all down---it kills off the last remaining bit of Cael code and paves the way for a better, easier to use, and more mechanically sound world---one where cash and "economy" means something, rather than just a silly "watch numbers go up" simulator.

Related changes:
- Start off with $500 of spacecash in your backpack
- Purges the "merch" store that's near cargo; most of the items have been moved to the arcade prize vendor. The candles have not been replaced---you can order them via cargo---the pipe has been made a premium item in vendors
- Slot machines are free play and give spacecash when you win them.
- The broken claw machine games are free to play now...not *that* matters, given they've been broken for months now
- All items in all vending machines are now free; premium items still require a coin
- Removes all traces of job objectives, as these were poorly handled and largely existed to pay people money
- A few instruments added to the instruments crate (and besides you can craft these anyways)
- Adds the flags to an assorted flags crate

unrelated tweak:
- Standardizes the date to be current year + 540 years, instead of +544, in line with other servers
  - This is just easier to remember and calculate 

The reason for purging the merch machine is twofold; it's a snowflakey, non TGUI interface that charges you cash for items that are unlimited in nature---everything else that you previously used money with that dispensed products was a vending machine. The other reason is to set players on the path of accepting that more things are going to be locked behind cargo or revolve around them, in the future.

Why remove this now, even if we don't wind up getting the economy for months? We're not out of anything. The economy, as it stands, is literally worthless---it doesn't function to really limit anything, mechanically, nor is it utilized for RP, beyond a, once in a blue moon "hey, I'll bribe you with this cash" (which is still something you'll be able to do, spacecash is intact). Even if we're without the current iteration for 6+ months (I hope it's not that long, but who knows), nothing *really* changes.

:cl: Fox McCloud
del: Sompletely deletes the entire economy and burns it all to the ground. It will be reborn, anew, some day
tweak: Start off with $500 in your backpack
tweak: Order flags from a flags crate instead of a merch computer
tweak: Order the D&D items and dice from cargo instead of a merch computer
tweak: Most merch computer odds and ends have been moved to arcade prizes
tweak: Smoking pipe is now a premium cigarette machine item
tweak: All vending machine items are free
tweak: Slot machines are free to spin and give out cash
tweak: Current date, in SS13 years, is current year + 540 years instead of +544
/:cl:

### CODE REVIEW

File: code/datums/supplypacks.dm [hunk 1]
```diff
@@ -1713,10 +1714,39 @@ GLOBAL_LIST_INIT(all_supply_groups, list(SUPPLY_EMERGENCY,SUPPLY_SECURITY,SUPPLY
...
+	containername = "flag assortment Crate"
+
+/datum/supply_packs/misc/dnd
+	name = "DND Crate"
+	contains = list(/obj/item/storage/box/characters,
+					/obj/item/storage/box/characters,
+					/obj/item/storage/box/dice,
+					/obj/item/storage/box/dice)
+	cost = 5
+	containername = "flag assortment Crate"
```
Comment (Mitchs98) [inline code/datums/supplypacks.dm:1748]: Just a heads up. You named the DND crate the same as the flag assortment crate.

Comment (Fox-McCloud) [reply code/datums/supplypacks.dm:1748]: oops

**Comment by Varghh** [comment]
I'm a bit sad about the Merch machine moving to the prize ticket, I used to buy the DnD figure sets and place one figure inside paper bags with food as the chef like the happy meal toy.

**Comment by Gatchapod** [comment]
Our current economy doesn't work or even make sense. True.
However, you can use it for small fun things with people who are fine with pretending credits matter.

For that reason, just this once I'll shout "improve, don't remove".
> It's a complicated system, and ripping this out while simultaneously putting that in would be extremely difficult and hard to analyze, especially since a lot of variables have similar names and behaviors.

Ultimately, I feel like this is the strongest argument supporting removal right now. However, you're talking about possible months before the follow-up PR introducing a new economy.
Would it really be a problem if we kept the economy as it is until you, or someone else, picks up the task of porting TG one?

Basically, I'm questioning timing, not actual removal. Feels like we'd lose nothing by keeping it around until someone starts working on the TG port.

**Comment by S34NW** [comment]
How will this affect traders?

**Comment by procdrone** [comment]
Burn it down, good riddance.

> 
> How will this affect trader events?

It will not, because space$ is not removed.

**Comment by Carthusia** [comment]
I agree with Gatcha, im fine with a revamp of the economy and all to make things worth while and make money matter, but why not wait until a replacement is ready or near ready before removing the old one if it could take potentially almost a year to acquire the replacement.

**Comment by datlo** [comment]
Looking over the code, this PR removes ATM machines and accounts (Good!). However, the issue is that most people will no longer have access to cash for RP purposes/traders other than by farming slots (And mining vendor? I think?)

Perhaps, as a stand-in for a proper economy to come, we could give people physical cash at roundstart, as a replacement for their account. That way we still have money for bribes/RP/traders/clown thieves, and tbh it'll be easier to manage and play with if everybody has physical credits roundstart.

**Comment by Fox-McCloud** [comment]
>Perhaps, as a stand-in for a proper economy to come, we could give people physical cash at roundstart, as a replacement for their account. That way we still have money for bribes/RP/traders/clown thieves, and tbh it'll be easier to manage and play with if everybody has physical credits roundstart.

I was going back and forth about this, but I'm more than happy to do this, as a stand-in, for now.

**Comment by Spacemanspark** [comment]
Not particularly a fan of removing the merch console, even if it's contents are being moved to the prize vendors. Is there no other alternative? 

Outside of that I'm not terribly keen on gutting all of this until the port is made. Why make a void like this, especially if there are plans to change it to the tg system? Mind, I'm not opposed to the tg system, I've heard it's economy system is quite something--but I want it to actually be in the game.

**Comment by thatdanguy23** [comment]
wait until the port is ready so we don't have to be without an economy for several years. 

also why don't we just rebase our code to TG at this point

**Comment by Mitchs98** [comment]
Honestly don't feel too strongly either way, though I do share sentiments about the merch console and its items being removed. At the same time it probably would be best to wait until work on the alternative starts/the alternative is actually fully fleshed out and approved even without work starting on it yet. 

Unless the current implementation has large performance issues or some such I don't see a reason to remove everything. Though at the same time I don't see many drawbacks.

**Comment by SabreML** [comment]
Just gonna point that this is PR #15555, which is pretty cool.

**Comment by Fox-McCloud** [comment]
We don't really lose anything by excising the economy, as it currently stands, and going with @datlo's suggestion.

Currently, bank accounts and such aren't really that impactful or utilized; no one is running out of cash buying their food---the only actual money sink is the store, and that's not really an issue either, especially since prices got lowered on it.

What cash is sometimes used for is random tips or exchanges or the traders---by everyone starting with cash, this interaction is still maintained, while everything else, that has zero impact, is excised.

As for the whole "improve don't remove"--that's kinda what's being aimed for here---what's being burnt down is unsalvageable and an utter mess, to make way for a system that *does* have interact with players.

Transitioning to spawning in with cash preserves the RP interactions people love and hold dear, while a bunch of *terrible* code gets preserved, and, functionally, practically nothing changes.

**Comment by Carthusia** [comment]
How much would people start with if we went with this immediately instead of waiting?

I like to run my own shops quite often and people having a decent amount of money is a big reason as to why it runs smoothly.

I really do look forward to a more concrete economy that has some persistence to it, I'm just afraid people having far to little would negatively affect their ability to RP in these kind of circumstances until the new system is put into place.

It would also still be nice to simply buy the items from the merch store instead of having to play arcade games that, in all honesty, stopped being fun the 4th time I played them. Similar to the slow floor ruin in lavaland. Good for a laugh for the first few times, but it gets old very quick.

Being forced to grind for a flag by playing Orion Trail round after round I feel would negate their usage in many cases, as I feel many people would stop bothering to use them after a short time.

**Comment by PidgeyThePirate** [comment]
I'd say give each person space cash roughly equivalent to half of what they would have had under old economy.

Also, what of the EFTPOS machines? They're everywhere.

**Comment by Spacemanspark** [comment]
@FoxMcCloud The main concerns I'm seeing here--and share, myself--is having the actual new system implemented instead of just gutting it and waiting, alongside the merch console removal.

Also, I've just noticed the the change in lore date you've placed, which is hardly covered in the PR. While we might not be a very lore driven server, forcing our lore to be constricted to other servers because it's 'easier' doesn't particularly sit well with me.

**Comment by S34NW** [comment]
Personally, I think the lore date change is fine. It's not like the date is a huge issue and having a round number is easier for quick maths

**Comment by Spacemanspark** [comment]
There's a few underlying issues I have with changing the date--but I will certainly say that it's not hard to look at the wiki or in game and see the current year. You don't need math for anything.

This also sets a precedent I'm not really pleased with; that being one person having the ability to unravel or modify large aspects of the game's lore (yes, the year is a large aspect) without any particular reason. Mechanically, nothing changes, so why touch it? Why do we need to be so 'consistent' with other servers?

**Comment by PidgeyThePirate** [comment]
I don't mind gutting out the current economy. It's basically worse than nothing because nobody ever has to interact with it and it takes up design space that could be used for a better system.

**Comment by Mitchs98** [comment]
> There's a few underlying issues I have with changing the date--but I will certainly say that it's not hard to look at the wiki or in game and see the current year. You don't need math for anything.
> 
> This also sets a precedent I'm not really pleased with; that being one person having the ability to unravel or modify large aspects of the game's lore (yes, the year is a large aspect) without any particular reason. Mechanically, nothing changes, so why touch it? Why do we need to be so 'consistent' with other servers?

Definitely agree with this. Especially for people who have set backstories for their characters already, yes people actually do that. I see no reason our lore date needs to change to accommodate the overall PR. Nor how it's actually relevant to the PR at hand, it seems largely out of scope if anything.

**Comment by datlo** [comment]
I agree with the year change point raised above. Even though I don't personally mind the date being changed that way. I think it should be its own PR.

**Comment by Kiyahitayika** [comment]
Fox McCloud seems to take on big projects and achieve results. If Fox thinks the current code is complicated and it's going to be easier to remove it and start from the ground-up, I believe it. 

>I don't mind gutting out the current economy. It's basically worse than nothing because nobody ever has to interact with it and it takes up design space that could be used for a better system.

Yep. @PidgeyThePirate put it well. It's hard to go piecemeal through a big system and change it. Sounds like Fox has a plan and knows what he wants to do. Removing the old system to make room for that makes sense from a coding perspective. Additionally, everything of use (vending machine items) will be free until the new system is implemented. 

Any chance a temporary system could be created using the points-based system of the mining vendor? Or people could use a bartering system for RP purposes.

**Comment by thatdanguy23** [comment]
> Fox McCloud seems to take on big projects and achieve results. If Fox thinks the current code is complicated and it's going to be easier to remove it and start from the ground-up, I believe it.

nobody's doubting this, the problem is the time it takes between implementations and the fact this isn't going to be the first nor the last time the github will likely be frozen while we wait several years for a feature ported from /tg/.
giving everyone a vague "oh i'll have it ready in a few months so let's remove it now" does not inspire confidence. have the port ready _first_.

**Comment by Dumbdumn5** [comment]
I was looking over this PR on the sheet the other day and I was going to raise a few concerns related to discussion above.

To be fair it _does_ often take a bit for larger PRs to come in, and though it's not often used, having a little bit of econ for edge-cases is at least a little neat. If it's completely broken I agree gutting it and putting in an alternative is a great idea and a rework should come along! That said, I can also see how removing it without that alternative coming in either in the same PR or shortly after can be frustrating. 

What'd you have in mind for the economy after the fact and how quickly after this PR would you consider that PR would be coming along? If it's longer than a month or so, why isn't it possible to include the code removing references to the old econ system you've already written here in a PR replacing it with the new and improved system instead of compensating for the gap in PR merges with the tweaks you've made here?

I understand if it's outside of the scope of this PR, but at least having the alternative system up at the same time as the removal similar to how Biotypes was a precursor to the Physiology PR would probably do a lot to help out with the idea that we've got things ready and they won't be sitting in feature limbo for a longer while waiting on code to come in!

**Comment by Mitchs98** [comment]
> > Fox McCloud seems to take on big projects and achieve results. If Fox thinks the current code is complicated and it's going to be easier to remove it and start from the ground-up, I believe it.
> 
> nobody's doubting this, the problem is the time it takes between implementations and the fact this isn't going to be the first nor the last time the github will likely be frozen while we wait several years for a feature ported from /tg/.
> giving everyone a vague "oh i'll have it ready in a few months so let's remove it now" does not inspire confidence. have the port ready _first_.

Basically my thoughts in a nutshell. I have no doubts it'll happen eventually and be done to a good degree. It's the meantime that bothers me.

**Review by Kyep — COMMENTED** [review]
I'll sign off on this.
While I had my own plan that would not necessarily follow TG, I had no firm date for implementing that plan, and axing the existing god-awful economy would be necessary either way.
Fox is right that the current economy system is useless. It adds nothing but pointless complexity. Good riddance to it.

**Review by Kyep — APPROVED** [review]

**Comment by Fox-McCloud** [comment]
Everyone should start with $500 in their backpack; more than enough for day-to-day RP interactions.

**Comment by Denthamos** [comment]
I would love to see a proper economy in the future and this is simply the first step to be taken.  While having it all done in one go would be nice, it is simply easier to do it in a step process for our coders.

I personally love VG's economy where the crew start off with a small amount in their accounts and then get paid every 15 minutes a wage based on their job.  Meanwhile certain areas on station like cargo, medical, science, and the chef have vendors they can put items into with a price to sell them at.  That money can then be put into departmental/personal accounts for purchases at cargo/traders which runs off money and not points.

Does this PR also wipe/remove the departmental accounts and station account which all start with a huge amount of money at round start?

**Comment by Fox-McCloud** [comment]
this will rise again, but for now, there's more important things.
